File: models/ipdmanagement/IPDDoctorAssignment.js
// models/IPDDoctorAssignment.js
import mongoose from "mongoose";

const IPDDoctorAssignmentSchema = new mongoose.Schema(
  {
    hospitalId: { type: mongoose.Schema.Types.ObjectId, required: true },
    branchId: { type: mongoose.Schema.Types.ObjectId, required: true },

    ipdAdmissionId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "IPDAdmission",
      required: true
    },

    doctorId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true
    },

    role: {
      type: String,
      enum: ["Primary", "Consultant", "Surgeon"],
      default: "Primary"
    }
  },
  { timestamps: true }
);

export default mongoose.model("IPDDoctorAssignment", IPDDoctorAssignmentSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/IPDDoctorAssignment.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "ipd_doctor_assignments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class IPDDoctorAssignment extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ipd_admission_id", nullable = false)
    private IPDAdmission ipdAdmission;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_id", nullable = false)
    private User doctor;

    @Enumerated(EnumType.STRING)
    private Role role = Role.Primary;

    public enum Role {
        Primary, Consultant, Surgeon
    }
}
